pkg/kiln/deploy_check.go

```diff
@@ -0,0 +1,106 @@
+package kiln
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/labels"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// ClusterConfig is the config used to interact with Shipyard's in-cluster artifacts
+type ClusterConfig struct {
+	Client       *kubernetes.Clientset
+	OrgLabelName string
+	AppLabelName string
+}
+
+// NewClusterConfig creates a new k8s in-cluster client
+func NewClusterConfig() (*ClusterConfig, error) {
+	clusterConfig := &ClusterConfig{}
+
+	clusterConfig.OrgLabelName = os.Getenv("ORG_LABEL")
+	if clusterConfig.OrgLabelName == "" {
+		return nil, fmt.Errorf("Missing required ORG_LABEL environment variable")
+	}
+
+	clusterConfig.AppLabelName = os.Getenv("APP_NAME_LABEL")
+	if clusterConfig.OrgLabelName == "" {
+		return nil, fmt.Errorf("Missing required APP_NAME_LABEL environment variable")
+	}
+
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	// creates the clientset
+	clusterConfig.Client, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clusterConfig, nil
+}
+
+// NewLocalClusterConfig creates a new k8s in-cluster client
+func NewLocalClusterConfig() (*ClusterConfig, error) {
+	clusterConfig := &ClusterConfig{}
+
+	clusterConfig.OrgLabelName = os.Getenv("ORG_LABEL")
+	if clusterConfig.OrgLabelName == "" {
+		return nil, fmt.Errorf("Missing required ORG_LABEL environment variable")
+	}
+
+	clusterConfig.AppLabelName = os.Getenv("APP_NAME_LABEL")
+	if clusterConfig.OrgLabelName == "" {
+		return nil, fmt.Errorf("Missing required APP_NAME_LABEL environment variable")
+	}
+
+	usr, err := user.Current()
+	if err != nil {
+		return nil, err
+	}
+
+	// uses the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", path.Join(usr.HomeDir, ".kube/config"))
```

Comment:
Kubernetes clientcmd provides a way of doing this automatically instead of generating when the kube config is by hand.  See: https://godoc.org/k8s.io/client-go/tools/clientcmd#BuildConfigFromFlags

It's the same code we use enrober and dispatcher I would vote we use the same logic in all three projects.  https://github.com/30x/dispatcher/blob/master/kubernetes/client.go#L54-L62 and https://github.com/30x/enrober/blob/staging/pkg/server/server.go#L97-L111.